plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibaseExecutionPlan.java

```diff
@@ -0,0 +1,178 @@
+        Class<? extends Connection> clazz = null;
+        Method method = null;
+        ExplainPlan expPlan = null;
+        
+        try {
+            conn = session.getOriginal();
+            clazz = conn.getClass();
+            
+            /* 
+             * There are two setExplain methods in Connction class: 
+             * The first one's argument is boolean, the second one's argument is byte.
+             * Here, the second method is required.
+             */
+            method = getMethod2SetExplainWithByteArgType(clazz, 
+                    setExplainPlan, AltibaseDataTypeDomain.BYTE.getTypeName().toLowerCase());
+
+            expPlan = AltibaseConstants.ExplainPlan.getByIndex(
+                    dataSource.getContainer().getPreferenceStore().getInt(
+                            AltibaseConstants.PREF_EXPLAIN_PLAN_TYPE));
+
+            method.invoke(conn, expPlan.getArgValue());
+            stmt = conn.prepareStatement(query);
+            
+            // In case of EXPLAIN_PLAN_ON, need to execute query
+            if (expPlan == AltibaseConstants.ExplainPlan.ON) {
+                if (query.trim().toUpperCase().startsWith("SELECT")) {
+                    stmt.getClass().getMethod("executeQuery").invoke(stmt);
+                } else {
+                    stmt.getClass().getMethod("execute").invoke(stmt);
+                }
+            }
+            
+            planQuery = (String) stmt.getClass().getMethod("getExplainPlan").invoke(stmt);
+
+        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException
+                | SecurityException | SQLException | ArrayIndexOutOfBoundsException e) {
+            log.error("Failed to execute explain plan: " + e.getMessage());
+        } finally {
+            if (stmt != null) {
+                try {
+                    stmt.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        return planQuery;
+    }
+
+
+    /**
+     * Returns a method (BYTE xxxx): BYTE data type argument of method with the given name. 
+     */
+    @SuppressWarnings("rawtypes")
+    private Method getMethod2SetExplainWithByteArgType(Class class1, String methodName, String argName) throws NoSuchMethodException {
+        for (Method method : class1.getMethods()) {
+            if (method.getName().equals(methodName)) {
+                for (Class paramType : method.getParameterTypes()) {
+                    if (paramType.toString().equals(argName)) {
+                        return method;
+                    }
+                }
+            }
+        }
+
+        throw new NoSuchMethodException(String.format("Unable to find the target method: [class] %s, [method] %s, [argument type] %s", 
+                        class1.getName(), setExplainPlan,  AltibaseDataTypeDomain.BYTE.getTypeName().toLowerCase()));
+    }
```

Comment:
You can query a method with a particular type signature using `getMethod`. Furthermore, we do have utilities that might assist you in `org.jkiss.utils.BeanUtils`

Reply:
Thanks a lot :) Though org.jkiss.utils.BeanUtils has good methods, Class.getMethod is more neat and specific.